spec/models/dataset_spec.rb

```diff
@@ -123,4 +123,19 @@
       expect(dataset.public_updated_at).to eq(dataset.updated_at)
     end
   end
+
+  describe "#released" do
+    it "returns true when there is a link" do
+      dataset = FactoryGirl.create(:dataset, datafiles: [FactoryGirl.create(:datafile)])
+      expect(dataset.released).to be_truthy
```

Comment:
How about we use a predicate matcher?

```rb
expect(dataset).to be_released

expect(dataset).not_to be_released
```

You could use RSpec's `context` to describe the different way `released` behaves:

```rb
context 'when there are datafiles' do
  let(:dataset) { FactoryGirl.create(:dataset, datafiles: [FactoryGirl.create(:datafile)]) }

  it { is_expected.to be_released }
end
```

Reply:
I like to avoid using contexts unless they actually add context; in this case, we're just changing the subject. It also results in more indentation meandering as you try to read the file. I used to do contexts a lot, but then found I was typing a lot of boilerplate for no benefit.

With the predicate matcher, I'm tempted but I'm thinking it looks a bit magic for someone coming to read the tests - unless you're very familiar with RSpec, using a smaller vocabulary is more friendly to the reader.